bids-validator/src/schema/applyRules.ts

```diff
@@ -441,31 +456,30 @@ function evalJsonCheck(
     if (severity && severity !== 'ignore' && !(keyName in json)) {
       if (requirement.issue?.code && requirement.issue?.message) {
         context.dataset.issues.add({
-          key: requirement.issue.code,
-          reason: requirement.issue.message,
+          code: requirement.issue.code,
+          subCode: keyName,
+          location: context.path,
           severity,
-          files: [{ ...context.file }],
+          codeMessage: requirement.issue.message,
+          rule: schemaPath,
         })
```

Comment:
Moving `codeMessage` out. Let deno reformat.

```suggestion
          rule: schemaPath,
        }, requirement.issue.message)
```